### PR TITLE
WalletTool: new option for setting the type of filter for syncing

### PR DESCRIPTION
**Pull request by schildbach**
Currently, only server-side (bloom) filtering is supported.

Child of #3307

### CODE REVIEW

**Review by msgilligan — CHANGES_REQUESTED** [review]

File: wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java [hunk 1]
```diff
@@ -323,6 +325,12 @@ public enum WaitForEnum {
         BALANCE
     }
 
+    public enum Filter  {
+        NONE,
+        SERVER, // bloom filter
```
[the remark this answers is not in the document]
Comment (schildbach) [reply wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java:330]: For user, the actual filter algorithm is not that important. What counts is if it is a client-side or a server-side filter, with impact on performance and privacy. Nowadays there are BIPs for both and I assume at some point we will implement client-side filtering as well.

File: wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java [hunk 2]
```diff
@@ -323,6 +325,12 @@ public enum WaitForEnum {
...
+        /* CLIENT */
```
[the remark this answers is not in the document]
Comment (schildbach) [reply wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java:331]: See above.

File: wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (schildbach) [reply wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java:330]: Can this be resolved?

File: wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (schildbach) [reply wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java:331]: Can this be resolved?

[the remark this answers is not in the document]
Comment (msgilligan) [reply wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java:331]: I still think `BLOOM` would be more self-explanatory to the majority of our users/developers. But `SERVER` is fine, too.

**Review by msgilligan — APPROVED** [review]
Approved. (Though I think `BLOOM` would be more self-explanatory to our intended audience than `SERVER`)

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**